images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -170,6 +202,19 @@ function run() {
  }
 
 if [[ "${EGRESS_DNS_PROXY_MODE:-}" == "unit-test" ]]; then
+
+    declare -A DNS_SERVERS=()
+    if [[ "${EGRESS_DNS_SERVERS+isset}" ]]; then
+        servers=(${EGRESS_DNS_SERVERS//;/ })
+        for server in "${servers[@]}"; do
```

Comment:
You have a test failure:

    --- FAIL: TestHAProxyFrontendBackendConf (0.02s)
    	egress_dns_proxy_test.go:202: test 1 unexpected error exit status 1, output: "./egress-dns-proxy.sh: line 221: servers[@]: unbound variable\n"

I think you need to use `${EGRESS_DNS_SERVERS+isset}` in the `if` statement in order not to fail when `EGRESS_DNS_SERVERS` is defined but empty.

Reply:
I mean, you need to change `${EGRESS_DNS_SERVERS+isset}` to ` ${EGRESS_DNS_SERVERS:+isset}`.